urllib3/connection.py

```diff
@@ -260,9 +266,40 @@ def connect(self):
                             or self.assert_fingerprint is not None)
 
 
+class SOCKSHTTPConnection(HTTPConnection):
+    """
+    Identical to :class:`HTTPConnection`, except overrides :meth:`socket.create_connection` with
+    :meth:`socks.create_connection`, using ``socks_opts`` arguments from ``**conn_kw`` to
+    construct the :class:`socks.socksocket` object.
+    """
+    def __init__(self, *args, **kw):
+        self.socks_opts = kw.pop('socks_opts', None)
+        super(SOCKSHTTPConnection, self).__init__(*args, **kw)
+
+    def _create_connection(self, address, timeout, **kw):
```

Comment:
Might be time to add an interface for providing a create_connection callable, rather than making every protocol adjustment override these methods.